src/TextBoxSearch/components/TextBoxSearchContainer.ts

```diff
@@ -0,0 +1,152 @@
+import { Component, ReactElement, createElement } from "react";
+import { findDOMNode } from "react-dom";
+import * as dijitRegistry from "dijit/registry";
+import * as dojoConnect from "dojo/_base/connect";
+import * as classNames from "classnames";
+
+import { Alert } from "../../Shared/components/Alert";
+import { DataSourceHelper, ListView } from "../../Shared/DataSourceHelper/DataSourceHelper";
+import { SharedUtils } from "../../Shared/SharedUtils";
+
+import { TextBoxSearch, TextBoxSearchProps } from "./TextBoxSearch";
+import { Validate } from "../Validate";
+
+interface WrapperProps {
+    class: string;
+    style: string;
+    friendlyId: string;
+    mxform?: mxui.lib.form._FormBase;
+    mxObject: mendix.lib.MxObject;
+}
+
+export interface ContainerProps extends WrapperProps {
+    attributeList: SearchAttributes[];
+    defaultQuery: string;
+    entity: string;
+    placeHolder: string;
+}
+
+export interface SearchAttributes {
+    attribute: string;
+}
+
+export interface ContainerState {
+    alertMessage?: string;
+    listViewAvailable: boolean;
+    targetListView?: ListView;
+    targetNode?: HTMLElement;
+    validationPassed?: boolean;
+}
+
+export default class SearchContainer extends Component<ContainerProps, ContainerState> {
+    private dataSourceHelper: DataSourceHelper;
+    private navigationHandler: object;
+
+    constructor(props: ContainerProps) {
+        super(props);
+
+        this.state = {
+            alertMessage: Validate.validateProps(this.props),
+            listViewAvailable: false
+        };
+
+        this.applySearch = this.applySearch.bind(this);
+        this.navigationHandler = dojoConnect.connect(props.mxform, "onNavigation", this, this.connectToListView.bind(this));
+    }
+
+    componentDidMount() {
+        const filterNode = findDOMNode(this).parentNode as HTMLElement;
+        const targetNode = SharedUtils.findTargetNode(filterNode);
+        if (targetNode) {
+            DataSourceHelper.hideContent(targetNode);
+        }
+    }
+
+    componentDidUpdate(_prevProps: ContainerProps, prevState: ContainerState) {
```

Comment:
no acronyms... `_previousProps` and `previousState`

Reply:
Done